### DISCUSSION TITLE
Using API in dataviewjs

### DISCUSSION THREAD

**Discussion by ramon86** [chhoumann/quickadd#135] [Q&A] [answered]
I hope this is something that's easy to achieve, but I cannot figure out how to use QuickAddAPI in other javascript contexts (in my case within a dataviewjs block).

The examples all work with it being available from the params object that's passed when you use actual QuickAdd macros, but is it possible to reference it on other ways as well?

**Comment by chhoumann** [comment]
Hey @ramon86! Currently, this isn't easily possible. The API is not 'public', making interaction with it through other plugins hard.
I've created an issue with a feature request for this #138. Will get to it when I can. :)

**Comment by ramon86** [comment]
Check, thanks for the explanation and feature request creation :)

**Comment by chhoumann** [comment, marked as answer]
Added in 0.8.0

**Comment by chhoumann** [comment]
Closing this out - the public API shipped back in 0.8.0, and it still works the same way from a dataviewjs block today:

```js
const quickAddApi = app.plugins.plugins.quickadd.api;
```

The API is documented here now: https://quickadd.obsidian.guide/docs/QuickAddAPI
